apps/ui/src/networks/offchain/actions.ts

```diff
@@ -161,40 +162,31 @@ export function createActions(
 
       const name = strategiesNames[0];
       const strategy = getOffchainStrategy(name);
-      if (!strategy) return [{ address: name, value: 0n, decimals: 0, token: null, symbol: '' }];
-
-      const result = await strategy.getVotingPower(
-        spaceId,
-        voterAddress,
-        strategiesOrValidationParams,
-        snapshotInfo
-      );
-
-      if (strategy.type !== 'remote-vp') {
-        return [
-          {
-            address: strategiesNames[0],
-            decimals: 0,
-            symbol: '',
-            token: '',
-            chainId: snapshotInfo.chainId,
-            value: result[0]
-          }
-        ];
+      let result: bigint[] = [];
+
+      if (!strategy || !isAddress(voterAddress)) {
+        result = Array(strategiesNames.length).fill(0n);
+      } else {
+        result = await strategy.getVotingPower(
+          spaceId,
+          voterAddress,
+          strategiesOrValidationParams,
+          snapshotInfo
+        );
       }
 
       return result.map((value: bigint, index: number) => {
         const strategy = strategiesOrValidationParams[index];
-        const decimals = parseInt(strategy.params.decimals || 18);
+        const decimals = parseInt(strategy.params?.decimals || 18);
 
         return {
-          address: strategy.name,
+          address: strategy.name || strategiesNames[index],
```

Comment:
This solves issue with address missing, but decimals will be wrong in this case, we only default to 18 for `remote-vp`, in other cases we return 1 or 0, so if we use 18 decimals it will be interpreted wrong.

To be honest I'd just bring the old special case `if` back, because with current approach it seems cleaner (no longer we have this mysterious if), but all those special cases are hidden within fallback values that are even more cryptic (and it's still not covering it entirely).